pkg/local_object_storage/blobstor/fstree/head.go

```diff
@@ -110,6 +110,9 @@ func (t *FSTree) extractHeaderAndStream(id oid.ID, f *os.File) (*objectSDK.Objec
 			if err != nil && !errors.Is(err, io.EOF) && !errors.Is(err, io.ErrUnexpectedEOF) {
 				return nil, f, fmt.Errorf("read full: %w", err)
 			}
+			if k == 0 {
+				return nil, f, fmt.Errorf("end of file reached before finding object: %w", io.ErrUnexpectedEOF)
```

Comment:
text just repeats the essence of `unexpected EOF`. We should either add text describing exactly this case, or simply `return io.ErrUnexpectedEOF`